packages/vega-statistics/src/regression/constant.js

```diff
@@ -0,0 +1,14 @@
+import { sum } from 'd3-array';
+import rSquared from './r-squared';
+
+export default function (data, x, y) {
+  const validData = data.filter(d => x(d) != null && y(d) != null),
+    meanY = sum(validData.map(y)) / validData.length,
```

Comment:
Overall the PR looks great. However, there is a minor discrepancy here with how we compute averages in the Aggregate transform. There, we use an online algorithm which should be more numerically stable than a standard sum-and-divide. We may want to be consistent throughout.

Reply:
Yes, NaNs should be considered "invalid" just like null/undefined. That would be consistent with Aggregate.